localstack/services/edge.py

```diff
@@ -74,6 +74,9 @@ def __init__(self, service_manager=None) -> None:
 
     def forward_request(self, method, path, data, headers):
 
+        if common.INFRA_STOPPED:
```

Comment:
Thats basically the core of the fix here. since infra is stopping trying to process any requests doesn't make sense anymore

Reply:
trying to understand the implications of this: could this potentially break user code that relies on polling of certain services that would, until now, still return something here?

Reply:
I mean it still returns something (503 response) so they at least shouldn't be blocked.

Also can't imagine a scenario right now where that would've ever lead to something useful since services are being stopped at the same time (which was leading to the race condition here since this was only the case in about 50% of runs locally).

@whummer  maybe you can also give your opinion on this. As I said I can't really see where user code would rely on things being (sometimes) still up after initiating an infra shutdown. :thinking:

Reply:
Hm, good point. I think it's fair to return 503 for external user requests after shutdown is initiated. Potentially we could still allow internal calls (determined via `is_internal_call` in line 167 below), in case there is cleanup logic going on that would rely on internal service calls. (Can't think of too many cases where this would be required, though, so not very critical in my view.)